examples/md-flexible/src/parsing/CLIParser.cpp

```diff
@@ -8,62 +8,80 @@
-                                         {MDFlexConfig::verletSkinRadiusStr, required_argument, nullptr, 'r'},
-                                         {MDFlexConfig::particlesSpacingStr, required_argument, nullptr, 's'},
-                                         {MDFlexConfig::tuningSamplesStr, required_argument, nullptr, 'S'},
-                                         {MDFlexConfig::traversalOptionsStr, required_argument, nullptr, 't'},
-                                         {MDFlexConfig::tuningStrategyOptionsStr, required_argument, nullptr, 'T'},
-                                         {MDFlexConfig::thermostatStr, required_argument, nullptr, 'u'},
-                                         {MDFlexConfig::verletRebuildFrequencyStr, required_argument, nullptr, 'v'},
-                                         {MDFlexConfig::vtkFileNameStr, required_argument, nullptr, 'w'},
-                                         {MDFlexConfig::vtkWriteFrequencyStr, required_argument, nullptr, 'W'},
-                                         {MDFlexConfig::selectorStrategyStr, required_argument, nullptr, 'y'},
-                                         {MDFlexConfig::yamlFilenameStr, required_argument, nullptr, 'Y'},
-                                         {MDFlexConfig::distributionStdDevStr, required_argument, nullptr, 'z'},
-                                         {nullptr, no_argument, nullptr, 0}};  // needed to signal the end of the array
+
+  static const std::vector<MDFlexConfig::MDFlexOptionInterface> relevantOptions{
+      config.newton3Options,
+      config.checkpointfile,
+      config.acquisitionFunctionOption,
+      config.cellSizeFactors,
+      config.boxLength,
+      config.containerOptions,
+      config.cutoff,
+      config.dataLayoutOptions,
+      config.deltaT,
+      config.dontCreateEndConfig,
+      config.tuningMaxEvidence,
+      config.functorOption,
+      config.dontMeasureFlops,
+      config.generatorOption,
+      config.iterations,
+      config.tuningInterval,
+      config.logLevel,
+      config.logFileName,
+      config.distributionMean,
+      config.maxTuningPhasesWithoutTest,
+      config.particlesPerDim,
+      config.particlesTotal,
+      config.relativeOptimumRange,
+      config.periodic,
+      config.tuningPhases,
+      config.verletClusterSize,
+      config.verletSkinRadius,
+      config.particleSpacing,
+      config.tuningSamples,
+      config.traversalOptions,
+      config.tuningStrategyOption,
+      config.useThermostat,
+      config.verletRebuildFrequency,
+      config.vtkFileName,
+      config.vtkWriteFrequency,
+      config.selectorStrategy,
+      config.yamlFilename,
+      config.distributionStdDev,
+      MDFlexConfig::MDFlexOption<std::string>("", "zsh-completions", false, 'Z', "Generate completions file for zsh."),
+      MDFlexConfig::MDFlexOption<std::string>("", "help", false, 'h', "Display this message."),
+  };
+
+  // TODO: verify getOptSwitchChars are unique
```

Comment:
todo

Reply:
done in 05c7ab99afc1508946c42fe5097f8a8e2fc49345